### ISSUE TITLE
429 Too Many Requests fails intermittendly

### ISSUE THREAD

**Issue by dkundel** [twilio-labs/serverless-toolkit#303]
We added exponential backoff but it seems to not actually pick up in some cases. Example for repro:

```
twilio serverless:init huge-project
cd huge-project/functions
echo "exports.handler = function(a, b, callback) { callback(null, 'hello')}" > demo{0..101}.js
cd ..
twilio serverless:deploy -l debug
```

Try running that for a couple of times and you should see it fail occasionally.

**Comment by cfluke-twilio** [comment]
I have a partner running into this as well. It is most commonly happening from the function version call here: https://github.com/twilio-labs/serverless-toolkit/blob/main/packages/serverless-api/src/api/functions.ts#L156
Should the library batch functions in groups? currently is looks like this use case is trying to update 50ish all at the same time causing the 429s.

**Comment by philnash** [comment]
It looks like that, but the concurrency should be controlled by the client using the limit function here: https://github.com/twilio-labs/serverless-toolkit/blob/main/packages/serverless-api/src/client.ts#L858.

There must be something else going on.

**Comment by ShelbyZ** [comment]
I had noticed this as well and had yet to attach a debugger, but I may suggest that we need to tune the concurrency limit (50 default) to a lower value. I tried to see if the docs for serverless mentioned any rate limits for service interaction, but could not find any.

By running your above example with `TWILIO_SERVERLESS_API_CONCURRENCY=10` I was able to deploy repeatedly. I will not suggest this as a final number, but if it is possible to get the actual from the team and then remove a fudge factor that may provide a better default to work with.
